exporter/datadogexporter/internal/testutils/test_utils.go

```diff
@@ -39,18 +39,20 @@ var (
 type DatadogServer struct {
 	*httptest.Server
 	MetadataChan chan []byte
+	// LogsData is the array of json requests sent to datadog backend
+	LogsData []map[string]interface{}
 }
 
 // DatadogServerMock mocks a Datadog backend server
 func DatadogServerMock(overwriteHandlerFuncs ...OverwriteHandleFunc) *DatadogServer {
 	metadataChan := make(chan []byte)
 	mux := http.NewServeMux()
 
+	server := &DatadogServer{}
```

Comment:
```suggestion
	var server DatadogServer
```
And you can just return `&server` at the end. It's more idiomatic. Also, why are you defining it here, you could just do it on the last line of this function:
```go
return &DatadogServer{
    Server: srv,
    MetadataChan: metadataChan
}
```